src/rgw/rgw_op.cc

```diff
@@ -2362,16 +2381,53 @@ void RGWGetObj::execute(optional_yield y)
     } catch (const buffer::error&) {}
   }
 
-
-  if (get_type() == RGW_OP_GET_OBJ && get_data) {
-    op_ret = handle_cloudtier_obj(attrs, sync_cloudtiered);
-    if (op_ret < 0) {
-      ldpp_dout(this, 4) << "Cannot get cloud tiered object: " << *s->object
-          <<". Failing with " << op_ret << dendl;
-      if (op_ret == -ERR_INVALID_OBJECT_STATE) {
-        s->err.message = "This object was transitioned to cloud-s3";
+  attr_iter = attrs.find(RGW_ATTR_MANIFEST);
+  if (attr_iter != attrs.end()) {
+    RGWObjManifest m;
+    try {
+    decode(m, attr_iter->second);
+    if (get_type() == RGW_OP_GET_OBJ && get_data && m.get_tier_type() == "cloud-s3") {
+      RGWObjTier tier_config;
```

Comment:
All cloud-tier related code including this if loop better be moved to handle_cloud_tier_obj() itself (like in the current code). it will make code in GetObj concise and will be better represented.